packages/cli/generation/ir-generator/src/converters/services/convertHttpResponse.ts

```diff
@@ -91,38 +85,25 @@ function convertJsonResponse(
     );
 }
 
-function getObjectPropertyFromTypeReference(
-    typeReference: TypeReference,
+function getObjectPropertyFromResolvedType(
+    resolvedType: ResolvedType,
     property: string,
     file: FernFileContext,
     typeResolver: TypeResolver
 ): ObjectProperty {
-    if (typeReference.type === "container" && typeReference.container.type === "optional") {
-        return getObjectPropertyFromTypeReference(typeReference.container.optional, property, file, typeResolver);
-    }
-    if (typeReference.type === "named") {
-        const resolvedType = typeResolver.resolveNamedTypeOrThrow({
-            referenceToNamedType: typeReference.name.originalName,
-            file,
-        });
-        switch (resolvedType._type) {
-            case "container":
-                return getObjectPropertyFromTypeReference(
-                    resolvedType.originalTypeReference,
-                    property,
-                    file,
-                    typeResolver
-                );
-            case "named":
-                if (isRawObjectDefinition(resolvedType.declaration)) {
-                    return getObjectPropertyFromObjectSchema(resolvedType.declaration, property, file, typeResolver);
-                }
-                throw new Error(
-                    "Internal error; response must be an object in order to return a property as a response"
-                );
-        }
+    switch (resolvedType._type) {
+        case "container":
+            if (resolvedType.container._type === "optional") {
+                return getObjectPropertyFromResolvedType(resolvedType.container.itemType, property, file, typeResolver);
+            }
+            break;
+        case "named":
+            if (isRawObjectDefinition(resolvedType.declaration)) {
+                return getObjectPropertyFromObjectSchema(resolvedType.declaration, property, file, typeResolver);
+            }
+            break;
     }
-    throw new Error(`Response does not have a property named ${property}.`);
+    throw new Error("Internal error; response must be an object in order to return a property as a response");
```

Comment:
@amckinney i missed this in the initiali review, but thoughts on handling every case and then adding an assertNever -- 

```
case "container": 
 return ...
case "optional":
  return ...
case "other1"
case "other2"
  throw new Error()
default: 
  assertNever();  
```
This will force us to handle the new cases if we ever add to the union

Reply:
Great call, added 👍